### DISCUSSION TITLE
question: does this work with nested subsections that are very long? Want to use it for digitzing medical regulations

### DISCUSSION THREAD

**Discussion by ottozastrow** [deepdoctection/deepdoctection#49] [Ideas]
cool!! In my use case PDF documents have several hundred pages. 
pdfs often have a hierarchy with sub-sections, sub-sub-sections, etc. 
would be super useful if it can do that. 
I'm working on a sideproject to make medical regulations more accessible. 
basically convert PDFs to a webpage, that has hyperlinks, good search, and is indexable by google.

**Comment by JaMe76** [comment]
Hi, super exiting project!

I worked on a POC for a fertilizer producer who had PDFs in various layout format akin to those you mentioned: Multi columns, sub and sub-sub sections tables with list in cells and so forth. You can definitely get very accurate results, if you iterate over the prediction outcome. 

The published models have been trained on medical research articles and therefore have a certain layout domain drift that has to be shifted towards your use-case. I used an annotation tool to review prediction result and re-label them, if necessary. Start with 30 samples but be selective with your samples and try to be as diverse as possible. Use different pdf documents to capture the full layout variety. I recommend not to invent any new labels as this will require a lot more samples to re-train a model. 
With 30 samples you can start fine-tuning the layout model and it will start improving very quickly. After fine tuning the second iteration starts with reviewing 30 samples etc. This will give you a decent layout detector. Saying that, after labeling e.g. 150-200 samples you can get an mAP/mAR increase up to 0.4. 

When it comes to building the pipeline you should be aware, what steps are going to be necessary, e.g.: 

1. Do I have PDF documents where characters can be extracted with a mining tool or are there embedded images where some OCR is required.
2. Do I need to perform table segmentation or do I just need some coarse layout detection?
3. Is reading order sensible? In some complex structured documents it is not clear at all how reading order should be determined and the only thing what matters is that layout blocks have to formed so that text passages/text lines can be grouped into one consistent reading entity.

**Reply by ottozastrow** [reply]
thanks for the detailed advice and starting points! You could add it to the readme I'm sure others will find it helpful too.
1. -> no OCR needed
2. -> yes there are some tables and some diagrams with text. But they're rare so the first version I won't care.
3. -> reading order is easy to determine since there are no columns. Sections are numbered and well formatted.
Your labeling approach sounds effective. Will do that, hard to anticipate the timeline though.
